commit/plugin.go

```diff
@@ -85,8 +101,130 @@ func NewPlugin(
 		reportCodec:       reportCodec,
 		reportingCfg:      reportingCfg,
 		chainSupport:      chainSupport,
-		observer:          observer,
+		merkleProcessor:   merkleProcessor,
+		tokenProcessor:    tokenprice.NewProcessor(),
+		feeProcessor:      chainfee.NewProcessor(),
+	}
+}
+
+func (p *Plugin) Query(_ context.Context, outCtx ocr3types.OutcomeContext) (types.Query, error) {
+	return types.Query{}, nil
+}
+
+func (p *Plugin) ObservationQuorum(_ ocr3types.OutcomeContext, _ types.Query) (ocr3types.Quorum, error) {
+	// Across all chains we require at least 2F+1 observations.
+	return ocr3types.QuorumTwoFPlusOne, nil
+}
+
+func (p *Plugin) Observation(
+	ctx context.Context, outCtx ocr3types.OutcomeContext, _ types.Query,
+) (types.Observation, error) {
+	prevOutcome := p.decodeOutcome(outCtx.PreviousOutcome)
+	fChain := p.ObserveFChain()
+	//TODO: Send fchain instead of computing it inside
```

Comment:
I think this is what you meant?
```suggestion
	//TODO: Move fchain to a new processor instead of computing it inside MerkleProcessor
```